app/models/response_snapshot.rb

```diff
@@ -0,0 +1,15 @@
+# Works same as Response.
+#
+# Uses the results stored in its ResponseFinished event instead of getting info realtime from R.
+class ResponseSnapshot < Response
```

Comment:
Een OOP stijl die gemaakt is voor static type checking heeft weinig zin. De ruby manier is via tests garanderen dat beide classes methodes bieden die tussen die twee overeen moeten komen. Stukken code die zowel een Response als een ResponseSnapshot kunnen verwachten moeten dus tests krijgen die in beide gevallen goed gaat, dan heb je het voldoende gegarandeerd en hoef je verder geen abstracte constructies in je code te hebben.